src/stores/room-list/algorithms/Algorithm.ts

```diff
@@ -163,18 +163,18 @@ export class Algorithm extends EventEmitter {
         this.recalculateActiveCallRooms(tagId);
     }
 
-    private updateStickyRoom(val: Room) {
-        this.doUpdateStickyRoom(val);
+    private async updateStickyRoom(val: Room) {
+        await this.doUpdateStickyRoom(val);
         this._lastStickyRoom = null; // clear to indicate we're done changing
     }
 
-    private doUpdateStickyRoom(val: Room) {
+    private async doUpdateStickyRoom(val: Room) {
```

Comment:
These cannot be async, this causes race conditions, please see git history around this file. You would need to introduce a semaphore to prevent operations racing each other.

Reply:
e.g. https://github.com/matrix-org/matrix-react-sdk/pull/6391

Reply:
Ouch. Will do, thanks!